### DISCUSSION TITLE
Slow route navigation

### DISCUSSION THREAD

**Discussion by codewithnazim** [vercel/next.js#65150] [App Router]
I am using lates app router, iam facing issues in route navigation, the navigation take to much time to navigate to other routes with Link and router imorted from next/navigation, is there any way to show some UI when page takes time to navigate , there is loading component in latest feature but its working only when the page is navigated on the route, I want loading on the current page if its take time to switch pages, please help I am stucked.

**Comment by SidKH** [comment]
Hi 👋

By default, the navigation will occur after the data on the route you navigate to is loaded.

If you want to speed up this app, you can add a `loading.js` or Suspense fallback for the route you navigate to. This way, the navigation will happen immediately showing the fallback from `loading.js` until the data is loaded.

https://github.com/vercel/next.js/assets/7762253/c4393593-ac3e-4f05-babd-232e48fc7edb

**Reply by codewithnazim** [reply]
So i need to add loading.js file in every route? i have used loading.js in the root of app.

**Reply by abdullah43577** [reply]
@codewithnazim do we need to add the loading.js to every route, I also have it in my root layout already.
I only noticed that it works in the root but not in deep nested layouts.
What did you find to work?
